app/assets/javascripts/editor/editor.js.erb

```diff
@@ -1019,6 +1022,25 @@ var CodeOceanEditor = {
         }
     },
 
+    applyChanges: function (delta) {
+        this.lastDeltaObject = delta;
+        this.editors[0].session.doc.applyDeltas([delta]);
+    },
+
+    compareDeltaObjects: function (delta, last_delta) {
+        if (delta === null || last_delta === null) {
+            return false;
+        }
+
+        const delta_data = delta.data
+        return delta_data.action === last_delta.action &&
+          delta_data.range.start.row === last_delta.range.start.row &&
+          delta_data.range.start.column === last_delta.range.start.column &&
+          delta_data.range.end.row === last_delta.range.end.row &&
+          delta_data.range.end.column === last_delta.range.end.column &&
+          delta_data.text === last_delta.text;
```

Comment:
I am aware that this is not ideal yet, but am fine to have it that way. Maybe just add a small comment above why we are making this very manual comparison :+1:.